vitrina/resources/migrations/0007_files_fix.py

```diff
@@ -0,0 +1,56 @@
+import mimetypes
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def fix_type_field_for_files(apps, schema_editor):
+    DatasetDistribution = apps.get_model("vitrina_resources", "DatasetDistribution")
+    Format = apps.get_model("vitrina_resources", "Format")
+
+    for resource in DatasetDistribution.objects.all():
+        if resource.file:
+            mime_type = mimetypes.guess_type(str(resource.file))
+            detected_format = Format.objects.filter(mimetype=mime_type).first()
+            resource.format = detected_format
+        if resource.type == 'URL':
+            url_format = Format.objects.filter(extension='URL').first()
```

Comment:
For URLs you actually need to do a real request and get format from response content type, like in example here:

```python
    status_code = 0
    content_type = ''
    try:
        resp = requests.get(url, stream=True, timeout=1)
    except requests.RequestException:
        pass
    else:
        if resp.status_code == 200:
            content_type = resp.headers.get('content-type')
            if ';' in content_type:
                content_type = content_type.split(';', 1)[0]
```

The you need to map received content type with a format:

```python
{
    'application/rdf+xml': 'RDF',
    'text/csv': 'CSV',
    'application/json': 'JSON',
    'application/geo+json': 'JSON',
    'application/xml': 'XML',
    'text/xml': 'XML',
    'application/vnd.ms-excel': 'XLSX',
    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet': 'XLSX',
    'text/html': 'HTML',
    'text/plain': 'URL',
    'application/zip': 'URL',
    '': 'URL',
}
```

And only fallback to URL if there is no way, to detect which format it actually is.

Reply:
By the way, I tink it is better to move this migration into a script, because this migration is most likely will take a long time to complete.